app/controllers/admin/teachers_controller.rb

```diff
@@ -4,17 +4,26 @@ class TeachersController < ResourceController
     def create
       super do |teacher|
         teacher.password = teacher.password_confirmation = SecureRandom.hex
-      end
+      end            
+      assign_role if role_params
```

Comment:
I believe this needs to go into the super block. we can discuss if it doesn't make sense